apps/cowswap-frontend/src/modules/twap/utils/getTwapOrderStatus.ts

```diff
@@ -37,3 +39,53 @@ export function isTwapOrderExpired(order: TWAPOrderStruct, startDate: Date | nul
 
   return nowTimestamp > endTime
 }
+
+export function getTwapParentStatusFromChildren(
+  childrenOrders: { status: OrderStatus; isCancelling?: boolean }[],
+): TwapOrderStatus {
+  if (!childrenOrders.length) return TwapOrderStatus.Pending
+
+  const hasOpen = childrenOrders.some((order) => order.status === OrderStatus.PENDING)
```

Comment:
Since there might be TWAP orders with thousands of children, we should keep the code productive.
Here we can make `.reduce` in order to make a map: `statuses: {[OrderStatus]: number}`
Then we can do smth like: `const hasOpen = statuses[OrderStatus.PENDING] > 0`